src/artemis/experiment_plans/full_grid_scan.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Callable
+
+from bluesky import plan_stubs as bps
+from dodal.devices.aperturescatterguard import ApertureScatterguard
+from dodal.devices.backlight import Backlight
+from dodal.devices.detector_motion import Det
+from dodal.devices.oav.oav_parameters import OAVParameters
+
+from artemis.experiment_plans.fast_grid_scan_plan import (
+    create_devices as fgs_create_devices,
+)
+from artemis.experiment_plans.fast_grid_scan_plan import get_plan as fgs_get_plan
+from artemis.experiment_plans.oav_grid_detection_plan import (
+    create_devices as oav_create_devices,
+)
+from artemis.experiment_plans.oav_grid_detection_plan import grid_detection_plan
+from artemis.log import LOGGER
+
+if TYPE_CHECKING:
+    from artemis.external_interaction.callbacks.fgs.fgs_callback_collection import (
+        FGSCallbackCollection,
+    )
+    from artemis.parameters.internal_parameters import InternalParameters
+
+detector_motion: Det = None
+backlight: Backlight = None
+aperture_scatterguard: ApertureScatterguard = None
+
+
+def create_devices():
+    fgs_create_devices()
+    oav_create_devices()
+    from artemis.experiment_plans.fast_grid_scan_plan import fast_grid_scan_composite
+
+    global detector_motion, backlight, aperture_scatterguard
+    detector_motion = Det("BL03I", name="detector_motion")
+    backlight = Backlight("BL03I-EA-BL-01:", name="backlight")
+    aperture_scatterguard = fast_grid_scan_composite.aperture_scatterguard
+    detector_motion.wait_for_connection()
+    backlight.wait_for_connection()
+    aperture_scatterguard.wait_for_connection()
+
+
+def wait_for_det_to_finish_moving(detector: Det, timeout=2):
+    LOGGER.info("Waiting for detector to finish moving")
+    SLEEP_PER_CHECK = 0.1
+    times_to_check = int(timeout / SLEEP_PER_CHECK)
+    for _ in range(times_to_check):
+        detector_state = yield from bps.rd(detector.shutter)
+        detector_z_dmov = yield from bps.rd(detector.z.motor_done_move)
+        LOGGER.info(f"Shutter state is {'open' if detector_state==1 else 'closed'}")
+        LOGGER.info(f"Detector z DMOV is {detector_z_dmov}")
+        if detector_state == 1 and detector_z_dmov == 1:
+            return
+        yield from bps.sleep(SLEEP_PER_CHECK)
+    raise Exception("Detector not finished moving")
```

Comment:
Should: This is much like the waiting for fast grid scan to be valid, e.g. messy and horrible. I think we need to put it in the device side probably? Certainly refactor it to be more Bluesky idiomatic